src/v/cluster/self_test_rpc_types.h

```diff
@@ -321,14 +341,15 @@ struct empty_request
 struct start_test_request
   : serde::envelope<
       start_test_request,
-      serde::version<0>,
+      serde::version<1>,
```

Comment:
This `serde::version<>` needs bumping to `serde::version<2>`, since `serde::version<0>` is the version of `start_test_request` with just `diskcheck_opts`, `netcheck_opts`, `serde::version<1>` is the version with `unknown_checks` added, and `serde::version<2>` needs to reflect the most recently added `cloudcheck_opts`.